### ISSUE TITLE
Can't @Mod inner classes?

### ISSUE THREAD

**Issue by SoniEx2** [MinecraftForge/FML#648]
Code:

```java
import cpw.mods.fml.common.Mod;
import cpw.mods.fml.common.event.FMLPreInitializationEvent;

/**
 * @author soniex2
 */
public class Broken {
    @Mod(modid = "test", name = "Test", version = "1.0")
    public static class TheMod {
        @Mod.EventHandler
        public void preInit(FMLPreInitializationEvent event) {
            System.out.println("Doesn't work");
        }
    }
}
```

If this is intended... Why? :/

**Comment by diesieben07** [comment]
This should be fixed in 1.8: https://github.com/MinecraftForge/FML/pull/617

**Comment by SoniEx2** [comment]
Oh, so that's why I wasn't able to find that minimal test case for MinecraftForge/MinecraftForge#1828... Thanks
